pkg/build/builder/cmd/builder.go

```diff
@@ -73,11 +71,8 @@ func newBuilderConfigFromEnvironment(out io.Writer, needsDocker bool) (*builderC
 		}
 		glog.V(4).Infof("redacted build: %v", string(bytes))
 	}
-	if errs := validation.ValidateBuild(cfg.build); len(errs) > 0 {
```

Comment:
Interesting.  What did you want this protection for?

Reply:
I think it was belt+suspenders.  It sort of confirmed that the build object we serialized into the env variable was in fact still valid when we deserialized it on the other side.  The build object would have already been through validation when it was created in the first place.  Given how impossible it would have been to retain this check(since validation runs against internal apis), I opted for removing the additional protection.

Reply:
> I think it was belt+suspenders. It sort of confirmed that the build object we serialized into the env variable was in fact still valid when we deserialized it on the other side. The build object would have already been through validation when it was created in the first place. Given how impossible it would have been to retain this check(since validation runs against internal apis), I opted for removing the additional protection.

Ok.  In general, client-side validation on something like this is questionable since the client rules are unlikely to consistently match the server rules.

Reply:
right, well prior to splitting things out it was using the same rules(same code).  but now it's being removed so it's moot.